projects/plugins/jetpack/extensions/blocks/calendly/calendly.php

```diff
@@ -114,7 +114,7 @@ function load_assets( $attr, $content ) {
 		$script  = <<<JS_END
 jetpackInitCalendly( '%s', '%s' );
 JS_END;
-		wp_add_inline_script( 'jetpack-calendly-external-js', sprintf( $script, esc_url( $url ), esc_js( $block_id ) ) );
+		wp_add_inline_script( 'jetpack-calendly-external-js', sprintf( $script, esc_url_raw( $url ), esc_js( $block_id ) ) );
```

Comment:
It seems like people find it hard to get escaping right for inline JS in WordPress code. Then again, WordPress's so-called "escaping" functions are pretty poor to start with.

* The old code was wrong in that it used `esc_url` which changes `&` to `&amp;` incorrectly for this context.
* Both the old and new are wrong in that they don't apply the escaping that's actually needed for inline JavaScript. If the URL happens to contain a `'` (rather than `%27`), things may break.
* The use of `esc_js()` for `$block_id` is wrong too, as that would also do the `&amp;` thing since it's intended for use in tag attributes like `onclick`. But since it's always alphanumeric that never comes up.

Probably what we really need here to do it right is something like this
```js
		$script  = <<<JS_END
jetpackInitCalendly( %s, %s );
JS_END;
		wp_add_inline_script(
			'jetpack-calendly-external-js',
			sprintf(
				$script,
				wp_json_encode( esc_url_raw( $url ), JSON_HEX_TAG | JSON_HEX_AMP ),
				wp_json_encode( $block_id, JSON_HEX_TAG | JSON_HEX_AMP )
			)
		);
```

Reply:
TIL! And you're completely correct on what's happening with the link. 

The problem is that your solution seems to result in this: `"https:\/\/calendly.com\/user\/30min?hide_event_type_details=0\u0026background_color=ac1e1e\u0026text_color=ff1c1c\u0026primary_color=16345f"`

Would you mind checking the changes which I've committed that handle those? Two things which I'm uncertain of though: 

1) I don't think we need to flag hex tags because that's already done when it's included in `add_query_arg`
2) As I understand it, can we not just use `esc_attr` for `$block_id`? Especially if there's no need to use `esc_js` for inline JavaScript. 

Thanks!! :)

Reply:
Here's a link for testing, by the way: https://calendly.com/sgdfkopfsdgpoksdfgkpodgfskpfsdkpdfsgdfg/30min (sorry - spam link that I just created for this)